drivers/clock_control/clock_stm32f2_f4_f7.c

```diff
@@ -50,6 +50,36 @@ uint32_t get_pllsrc_frequency(void)
 	return 0;
 }
 
+#if defined(STM32_CK48_ENABLED)
+/**
+ * @brief calculate the CK48 frequency depending on its clock source
+ */
+__unused
+uint32_t get_ck48_frequency(void)
+{
+	if (LL_RCC_GetCK48MClockSource(LL_RCC_CK48M_CLKSOURCE) ==
+			LL_RCC_CK48M_CLKSOURCE_PLL) {
+		/* Get the PLL Q freq. No HAL macro for that */
+		return __LL_RCC_CALC_PLLCLK_48M_FREQ(HSE_VALUE,
+						LL_RCC_PLLI2S_GetDivider(),
+						LL_RCC_PLLI2S_GetN(),
+						LL_RCC_PLLI2S_GetQ()
+						);
+	} else if (LL_RCC_GetCK48MClockSource(LL_RCC_CK48M_CLKSOURCE) ==
+			LL_RCC_CK48M_CLKSOURCE_PLLI2S) {
+		/* Get the I2S PLL Q freq. No HAL macro for that */
+		return __LL_RCC_CALC_PLLI2S_48M_FREQ(HSE_VALUE,
```

Comment:
This supposes that HSE is used, but HSI could also be used as PLL source clock.
Function RCC_PLLI2S_GetFreqDomain_48M could be used instead.

Reply:
could it be LL_RCC_PLLI2S_GetMainSource ?